app/actions/utils.ts

```diff
@@ -1,8 +1,29 @@
+import { AppNotificationStatus } from "state/app";
+
 export type TxnStatus =
   | "userConfirmation"
   | "networkConfirmation"
   | "done"
-  | "userRejected"
-  | "error";
+  | "error"
+  | "claimExceeded";
+
+export type OnStatusHandler = (status: TxnStatus, message: string) => void;
 
-export type OnStatusHandler = (status: TxnStatus) => void;
+export const getStatusMessage = (status: AppNotificationStatus) => {
+  const { statusType, message } = status;
+  if (status && statusType === "error" && message) {
+    if (message === "userRejected")
+      return "You chose to reject the transaction";
+    return message;
+  } else if (statusType === "userConfirmation") {
+    return "Please click 'confirm' in your wallet to continue.";
+  } else if (statusType === "networkConfirmation") {
+    return "Transaction initiated. Waiting for network confirmation.";
+  } else if (statusType === "error") {
+    return "❌ Error: something went wrong...";
```

Comment:
One of mine failed (never seen this error before, probably an RPC hiccup) and gave me this modal:

![image](https://user-images.githubusercontent.com/88635679/150046328-29f1d2d2-ab43-4f13-82ef-46be54f9c7ce.png)